taxcalc/tests/test_simpletaxio.py

```diff
@@ -0,0 +1,141 @@
+    '2 2013 0 2 0 1 15000 0    0 0 50000 70000 0 0 0 0 0 0 0 0    0 -3000\n'
+)
+REFORM_CONTENTS = """
+// Example of a reform suitable for use as an optional SimpleTaxIO reform file.
+// This JSON file can contain any number of trailing //-style comments, which
+// will be removed before the contents are converted from JSON to a dictionary.
+// The primary keys are policy parameters and secondary keys are years.
+// Both the primary and secondary key values must be enclosed in quotes (").
+// Boolean variables are specified as true or false (no quotes; all lowercase).
+{
+    "_AMT_tthd": // AMT taxinc threshold separating the two AMT tax brackets
+    {"2015": [200000],
+     "2017": [300000]
+    },
+    "_EITC_c": // maximum EITC amount by number of qualifying kids (0,1,2,3+)
+    {"2016": [[ 900, 5000,  8000,  9000]],
+     "2019": [[1200, 7000, 10000, 12000]]
+    },
+    "_II_em": // personal exemption amount (see indexing changes below)
+    {"2016": [6000],
+     "2018": [7500],
+     "2020": [9000]
+    },
+    "_II_em_cpi": // personal exemption amount indexing status
+    {"2016": false, // values in future years are same as this year value
+     "2018": true   // values in future years indexed with this year as base
+    },
+    "_SS_Earnings_c": // social security (OASDI) maximum taxable earnings
+    {"2016": [300000],
+     "2018": [500000],
+     "2020": [700000]
+    }
+}
+
+// The JSON above is translated into the following Python dictionary with
+// calendar year as the primary key, which is the data structure required
+// as input to the Policy class implement_reform(reform) method.
+//   reform = {
+//       2015: {
+//           '_AMT_tthd': [200000]
+//       },
+//       2016: {
+//           '_EITC_c': [[ 900, 5000,  8000, 9000]],
+//           '_II_em': [6000],
+//           '_II_em_cpi': False,
+//           '_SS_Earnings_c': [300000]
+//       },
+//       2017: {
+//           '_AMT_tthd': [300000]
+//       },
+//       2018: {
+//           '_II_em': [7500],
+//           '_II_em_cpi': True,
+//           '_SS_Earnings_c': [500000]
+//       },
+//       2019: {
+//           '_EITC_c': [[1200, 7000, 10000, 12000]]
+//       },
+//       2020: {
+//           '_II_em': [9000],
+//           '_SS_Earnings_c': [700000]
+//       }
+//   }
+"""
+
+
+@pytest.yield_fixture
+def input_file():
+    """
+    Temporary input file for SimpleTaxIO constructor.
+    """
+    ifile = tempfile.NamedTemporaryFile(mode='a', delete=False)
+    ifile.write(INPUT_CONTENTS)
+    ifile.close()
+    # must close and then yield for Windows platform
+    yield ifile
+
+
+@pytest.yield_fixture
+def reform_file():
+    """
+    Temporary reform file for SimpleTaxIO constructor.
+    """
+    rfile = tempfile.NamedTemporaryFile(mode='a', delete=False)
+    rfile.write(REFORM_CONTENTS)
+    rfile.close()
+    # must close and then yield for Windows platform
+    yield rfile
+
+
+def test_1(input_file):  # pylint: disable=redefined-outer-name
```

Comment:
@talumbau, Yes all the other test file use very long test_ function name and have no docstring.  This style, whatever its merits, drives pylint crazy.  Pylint assigns two warnings to each test_ function: one for too long a function name and one for the missing docstring.  The style here eliminates both warnings and provides unlimited space (not just an arcane, long function name) in the docstring to explain what is being tested.  If a test fails the programmer is going to have to look at the full code for the test anyway, so I don't see that a long function name is much of an advantage.  So, thanks for the suggestion, but I think I'll stick to the style in the test_simpletaxio.py file.